whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java

```diff
@@ -156,6 +161,80 @@ public Object mapSearch(final String requestUrl, final List<RpslObject> objects,
+            final RpslObject object = it.next();
+
+            switch (object.getType()) {
+                case AUT_NUM:
+                    autnums.add((Autnum) getRdapObject(requestUrl, object, objectDao.getLastUpdated(object.getObjectId()),
+                            Optional.empty()));
+                    break;
+                case INETNUM:
+                    ipv4Networks.add((Ip) getRdapObject(requestUrl, object, objectDao.getLastUpdated(object.getObjectId()),
+                            Optional.empty()));
+                    break;
+                case INET6NUM:
+                    ipv6Networks.add((Ip) getRdapObject(requestUrl, object, objectDao.getLastUpdated(object.getObjectId()),
+                            Optional.empty()));
+                    break;
+                case ORGANISATION:
+                    organisation = getRdapObject(requestUrl, object, objectDao.getLastUpdated(object.getObjectId()), Optional.empty());
+                    break;
+                default:
+                    throw new IllegalStateException("Incorrect object type " + object.getType());
+            }
+
+        }
+
+        organisation.setAutnums(autnums);
+        organisation.setIpv4(ipv4Networks);
+        organisation.setIpv6(ipv6Networks);
+
+        if (ipv4Networks.size() + ipv6Networks.size() + autnums.size()  > maxResultSize) {
+            final Notice outOfLimitNotice = new Notice();
+            outOfLimitNotice.setTitle(String.format("limited search results to %s maximum" , maxResultSize));
+            organisation.getNotices().add(outOfLimitNotice);
+        }
+        return mapCommons(organisation, requestUrl);
+    }
+
+    private Iterator<RpslObject> getTopLevelObjects(final Iterator<RpslObject> objects) {
+
+        final List<RpslObject> autnumRpsl = Lists.newArrayList();
+        final List<RpslObject> organisationRpsl = Lists.newArrayList();
+        final List<RpslObject> ipv4Rpsl = Lists.newArrayList();
+        final List<RpslObject> ipv6Rpsl = Lists.newArrayList();
+
+        objects.forEachRemaining(rpsl -> {
+            switch (rpsl.getType()){
+                case AUT_NUM:
+                    autnumRpsl.add(rpsl);
+                    break;
+                case ORGANISATION:
+                    organisationRpsl.add(rpsl);
+                    break;
+                case INETNUM:
+                    ipv4Rpsl.add(rpsl);
+                    break;
+                case INET6NUM:
+                    ipv6Rpsl.add(rpsl);
+                    break;
+                default:
+                    throw new IllegalStateException("Incorrect object type " + rpsl.getType());
+            }
+        });
+
+        return Stream.concat(Stream.concat(Stream.concat(autnumRpsl.stream(), organisationRpsl.stream()),
+                topLevelFilter.filter(ipv4Rpsl).stream()), topLevelFilter.filter(ipv6Rpsl).stream()).iterator();
+    }
```

Comment:
I think Toplevel filter uses IpInterval class you do not have to create a separate list and filter them separately for ipv4 and ipv6.